pkg/scheduler/eventhandlers.go

```diff
@@ -142,6 +142,18 @@ func (sched *Scheduler) updatePodInSchedulingQueue(oldObj, newObj interface{}) {
 		return
 	}
 
+	isUnschedPodUpdateIncluded := false
+	for _, evt := range framework.PodSchedulingPropertiesChange(newPod, oldPod) {
+		defer metrics.EventHandlingLatency.WithLabelValues(evt.Label).Observe(metrics.SinceInSeconds(start))
+		if evt.Label == framework.UnscheduledPodUpdate.Label {
+			isUnschedPodUpdateIncluded = true
+		}
+	}
+	if !isUnschedPodUpdateIncluded {
+		// make sure we always record the latency with UnscheduledPodUpdate label.
+		defer metrics.EventHandlingLatency.WithLabelValues(framework.UnscheduledPodUpdate.Label).Observe(metrics.SinceInSeconds(start))
+	}
+
```

Comment:
You don't need to use a variable:

```suggestion
    defer metrics.EventHandlingLatency.WithLabelValues(framework.UnscheduledPodUpdate.Label).Observe(metrics.SinceInSeconds(start))
	for _, evt := range framework.PodSchedulingPropertiesChange(newPod, oldPod) {
	    if evt.Label != framework.UnscheduledPodUpdate.Label {
		    defer metrics.EventHandlingLatency.WithLabelValues(evt.Label).Observe(metrics.SinceInSeconds(start))
		}
	}
```